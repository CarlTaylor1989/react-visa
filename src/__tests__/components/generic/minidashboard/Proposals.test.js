/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  Proposals,
  mapStateToProps
} from '../../../../components/generic/minidashboard/Proposals';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    wins: 2,
    losses: 26
  };
});

test('renders the Proposals component', () => {
  const wrapper = shallow(<Proposals {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('should mapStateToProps', () => {
  // setup
  const state = {
    clientProposalData: {
      clients: [
        {
          proposalCompleted: true,
          status: 2
        },
        {
          proposalCompleted: true,
          status: 2
        },
        {
          proposalCompleted: true,
          status: 4
        }
      ]
    }
  };

  const expected = {
    wins: 2,
    losses: 1
  };

  // function under test
  const props = mapStateToProps(state);

  // expectation
  expect(props).toEqual(expected);
});
