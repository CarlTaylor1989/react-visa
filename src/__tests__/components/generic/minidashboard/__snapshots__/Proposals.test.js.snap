// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Proposals component 1`] = `
<div
  className="proposals"
>
  <Translate
    tag="span"
    value="generic.miniDashboard.proposals"
  />
  <div
    className="winLossContainer"
  >
    <Translate
      tag="span"
      value="generic.miniDashboard.wins"
    />
    <div
      className="winsLosses"
    >
      2
      <Translate
        tag="span"
        value="generic.miniDashboard.proposalsDivider"
      />
      26
    </div>
    <Translate
      tag="span"
      value="generic.miniDashboard.losses"
    />
  </div>
</div>
`;
